scripts/thaliPrerequisite.js

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
+/*
+ This script will do the following tasks
+ 0. Clone the working jxcore-cordova plugin
+ 1. Fix issue on "can not replace existing file"
+ 2. Install the thali node module to global (until we get thali in npm repo)
+ 3. Move the build-extras.gradle to platform
+ 4: Create a dummy file to prevent future execution by other plugin add operation
+ */
+
+
+var fs = require('fs');
+var exec = require('child_process').execSync;
+
+
+var rootdir = process.argv[2];
+//Dummy file to prevent the execution of this script when the developer add other Cordova plugins
+var dummyFile = rootdir + '/../plugins/org.thaliproject.p2p/dummy.js';
+
+//Replace JXcoreExtension.java
+//Copy from \plugins\org.thaliproject.p2p\src\android\java\io\jxcore\node to \platforms\android\src\io\jxcore\node
+function replaceJXCoreExtension(){
+    var sourceFile =  rootdir + '/../plugins/org.thaliproject.p2p/src/android/java/io/jxcore/node/JXcoreExtension.java';
+    var targetFile = rootdir + '/../platforms/android/src/io/jxcore/node/JXcoreExtension.java';
+    fs.writeFileSync(targetFile, fs.readFileSync(sourceFile));
+}
+
+//Check whether this is the first time this script is getting executed or not
+//When ever this script run, it creates the dummyfile at the end of execution
+function isFirstTime(){
+    if (fs.existsSync(dummyFile))
```

Comment:
Wouldn't it be easier to just specify `!fs.existsSync(dummyFile)`? Now I personally admit that I don't like ! (I think it's hard to read) but you could also do `fs.existsSync(dummyFile) == false`. But in any case I believe that taking 4 lines for this seems excessive. I think it actually makes the code harder to read.